app/navigation/NavigatorForChangeActivity.scala

```diff
@@ -119,34 +120,53 @@ class NavigatorForChangeActivity @Inject() extends Navigator {
     case _ => _ => defaultCall
   }
 
-  private def navigationFollowingImports(userAnswers: UserAnswers): Call = {
-    userAnswers.get(AmountProducedPage) match {
-      case Some(AmountProduced.Large) => navigateForLargeAmountProducedFollowingImports(userAnswers)
-      case Some(AmountProduced.None) => navigateForAmountProducedNoneFollowingImports(userAnswers)
-      case Some(AmountProduced.Small) => defaultCall
+  private def navigationFollowingImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(AmountProducedPage) , mode) match {
+      case (Some(AmountProduced.Large), mode) => navigateForLargeAmountProducedFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.None ), mode) => navigateForAmountProducedNoneFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.Small), mode) => navigateForAmountProducedSmallProducerImports(userAnswers, mode)
       case _ => routes.AmountProducedController.onPageLoad(NormalMode)
     }
   }
 
-  private def navigateForAmountProducedNoneFollowingImports(userAnswers: UserAnswers): Call =
-    userAnswers.get(ContractPackingPage)
+  private def navigateForAmountProducedSmallProducerImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(SecondaryWarehouseDetailsPage), mode)
     match {
-      case Some(true) if userAnswers.packagingSiteList.isEmpty =>
+      case (Some(_), mode)  if mode == CheckMode   => routes.ChangeActivityCYAController.onPageLoad
```

Comment:
Any match case check where you have `mode) if mode == NormalMode/CheckMode` should be replaced with `NormalMode/CheckMode)` and removing the need for the if clause. Should make the case match neater. There's a few places to implement this further on aswell.

When that is implemented, I think it might help see if there are any further ways to neaten up the case matches!